library/core/functions.render.php

```diff
@@ -560,15 +560,7 @@ function hasEditProfile($userID) {
             return false;
         }
 
-        $result = checkPermission('Garden.Profiles.Edit') && C('Garden.UserAccount.AllowEdit');
-
-        $result &= (
-            C('Garden.Profile.Titles') ||
-            C('Garden.Profile.Locations', false) ||
-            C('Garden.Registration.Method') != 'Connect'
-        );
-
-        return $result;
+        return checkPermission('Garden.Profiles.Edit') && C('Garden.UserAccount.AllowEdit');
```

Comment:
My thinking here is that this check is only in place to allow a admin to turn profile editing back on using Locations or Titles. This check is not needed you are not disabling the edit profile functionality but rather only disabling the fields that will be managed by the single sign on.